### PR TITLE
feat: Migrate network container resource to the new Atlas SDK

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ [INTMDB-1095](https://jira.mongodb.org/browse/INTMDB-1095)

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Manual QA performed:

- [x] cfn invoke for each of CRUDL/cfn test
- [x] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [x] Published to AWS private registry
- [x] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [x] Deleted stack to ensure resources are deleted
- [x] Created multiple resources in same stack
- [x] Validated in Atlas UI
- [x] Included screenshots

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

### CFN test
```bash
cfn test --function-name TestEntrypoint --verbose

== 11 passed, 4 skipped, 9 deselected in 62.22s (0:01:02) ==
```

### CFN stack

```bash
atlas networking container ls --projectId 65156f486d71135a99a61af5 -o json                                                      16.18.0
[
  {
    "atlasCidrBlock": "10.8.2.0/24",
    "regionName": "US_EAST_1",
    "id": "65156f4e6d71135a99a61b2e",
    "providerName": "AWS",
    "provisioned": false
  }
]
```

Screenshot 2023-09-28 at 13 29 28

### CODE REVIEW

**Review by AgustinBettati — APPROVED** [review]
LGTM. Small detail we can improve is remove the resource.go file if we rename the Create/Update/Delete.. functions.

File: cfn-resources/network-container/cmd/resource/create.go [hunk 1]
```diff
@@ -41,19 +39,18 @@ func createOperation(req handler.Request, prevModel *Model, currentModel *Model)
...
+	client, peErr := util.NewAtlasClient(&req, currentModel.Profile)
 	if peErr != nil {
 		return *peErr, nil
 	}
 
-	containerRequest := &mongodbatlas.Container{}
-	containerRequest.RegionName = *currentModel.RegionName
-	containerRequest.ProviderName = constants.AWS
-	containerRequest.AtlasCIDRBlock = *currentModel.AtlasCidrBlock
+	aws := constants.AWS
```
Comment (marcosuma) [inline cfn-resources/network-container/cmd/resource/create.go:48]: any specific reason to define the variable here? Can we just use `constants.AWS`?

**Review by marcosuma — APPROVED** [review]

File: cfn-resources/network-container/cmd/resource/create.go (hunk 1 above)
Comment (andreaangiolillo) [reply cfn-resources/network-container/cmd/resource/create.go:48]: unfortunately no 😢 , `ProviderName` requires a pointer and you cannot use `&` with a constant

**Comment by andreaangiolillo** [comment]
> Small detail we can improve is remove the resource.go file if we rename the Create/Update/Delete.. functions.

@AgustinBettati Sure, Updated

File: cfn-resources/network-container/cmd/resource/create.go (hunk 1 above)
Comment (AgustinBettati) [reply cfn-resources/network-container/cmd/resource/create.go:48]: possible aleternative is to use admin.PtrString(constants.AWS)

**Review by AgustinBettati — DISMISSED** [review]

File: cfn-resources/network-container/cmd/resource/create.go (hunk 1 above)
Comment (andreaangiolillo) [reply cfn-resources/network-container/cmd/resource/create.go:48]: ah i was not aware of this function. Updating

Comment (andreaangiolillo) [reply cfn-resources/network-container/cmd/resource/create.go]: Updated

**Review by AgustinBettati — APPROVED** [review]

File: cfn-resources/network-container/cmd/resource/create.go [hunk 2]
```diff
@@ -21,18 +21,16 @@ import (
...
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
-	"go.mongodb.org/atlas/mongodbatlas"
+	"go.mongodb.org/atlas-sdk/v20230201008/admin"
 )
 
 var createRequiredFields = []string{constants.ProjectID, constants.RegionName, constants.AtlasCIDRBlock}
 
-func createOperation(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
 	if err := validateCreateModel(createRequiredFields, currentModel); err != nil {
```
Comment (lantoli) [inline cfn-resources/network-container/cmd/resource/create.go:34]: isn't there a setup method to configure the logger?

File: cfn-resources/network-container/cmd/resource/update.go [hunk 3]
```diff
@@ -19,48 +19,44 @@ import (
 	"fmt"
 
 	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/mongodb/mongodbatlas-cloudformation-resources/profile"
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/constants"
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
 	progressevents "github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
```
Comment (lantoli) [inline cfn-resources/network-container/cmd/resource/update.go:25]: I already fixed a problem in this package, can you try to remove the alias, replace ocurrences of progressevents by progressevent and see if make works?

**Pull request merged**